### PR TITLE
feat: remove query to backend to get expression width

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Progress on #4961 

## Summary\*

This PR removes the query to the backend in order to get the expression width, for now we just have a blanket default of ~4~ 3 (Looks like we're still waiting to update to a version of `bb` which uses 4). I've decided to split the ability to specify a package-level override of this default as I'd need to do some mucky stuff related to `compile_options`.

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.
We don't have docs related to the expression width afaik.
# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by vezenovm — APPROVED** [review]

**Comment by vezenovm** [comment]
Approved, just looks like we're missing some references to the backend info cmd

File: compiler/noirc_driver/src/lib.rs [hunk 1]
```diff
@@ -54,8 +54,8 @@ pub const NOIR_ARTIFACT_VERSION_STRING: &str =
 #[derive(Args, Clone, Debug, Default)]
 pub struct CompileOptions {
     /// Override the expression width requested by the backend.
-    #[arg(long, value_parser = parse_expression_width)]
-    pub expression_width: Option<ExpressionWidth>,
+    #[arg(long, value_parser = parse_expression_width, default_value = "4")]
```
Comment (TomAFrench) [inline compiler/noirc_driver/src/lib.rs:57]:
```suggestion
    #[arg(long, value_parser = parse_expression_width, default_value = "3")]
```

File: compiler/noirc_driver/src/lib.rs [hunk 2]
```diff
@@ -54,8 +54,8 @@ pub const NOIR_ARTIFACT_VERSION_STRING: &str =
 #[derive(Args, Clone, Debug, Default)]
 pub struct CompileOptions {
     /// Override the expression width requested by the backend.
-    #[arg(long, value_parser = parse_expression_width)]
-    pub expression_width: Option<ExpressionWidth>,
+    #[arg(long, value_parser = parse_expression_width, default_value = "3")]
```
Comment (TomAFrench) [inline compiler/noirc_driver/src/lib.rs:57]:
```suggestion
    #[arg(long, value_parser = parse_expression_width, default_value = "4")]
```

**Comment by TomAFrench** [comment]
Really not sure what's up with CI here. The `gates_report.sh` script runs just fine locally.

**Pull request merged**